x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/PhysicalVerifier.java

```diff
@@ -36,11 +35,6 @@ public Collection<Failure> verify(PhysicalPlan plan) {
         Failures depFailures = new Failures();
 
         plan.forEachDown(p -> {
-            if (p instanceof AggregateExec agg) {
-                var exclude = Expressions.references(agg.ordinalAttributes());
-                DEPENDENCY_CHECK.checkPlan(p, exclude, depFailures);
```

Comment:
Simplified by excluding the ordinal attributes already in `AggregateExec.references()`, which should be more correct, as these attributes are not required for the `AggregateExec` to be executed.